public/app/features/alerting/unified/state/AlertmanagerContext.tsx

```diff
@@ -45,13 +56,26 @@ const AlertmanagerProvider = ({ children, accessType }: Props) => {
   const defaultSource = GRAFANA_RULES_SOURCE_NAME;
 
   // queryParam > localStorage > default
-  const desiredAlertmanager = sourceFromQuery ?? sourceFromStore ?? defaultSource;
+  const desiredAlertmanager = alertmanagerSourceName ?? sourceFromQuery ?? sourceFromStore ?? defaultSource;
   const selectedAlertmanager = isAlertManagerAvailable(availableAlertManagers, desiredAlertmanager)
     ? desiredAlertmanager
     : undefined;
 
+  const selectedAlertmanagerConfig = getAlertmanagerDataSourceByName(selectedAlertmanager)?.jsonData;
+
+  // determine if we're dealing with an Alertmanager data source that supports the ruler API
+  const isGrafanaAlertmanager = selectedAlertmanager === GRAFANA_RULES_SOURCE_NAME;
+  const isAlertmanagerWithConfigAPI = CONFIG_API_ENABLED_ALERTMANAGER_FLAVORS.includes(
```

Comment:
As pointed out by @konrad147 we need to take another critical look at this assumption. The provisioned Alertmanager data sources in Hosted Grafana don't actually have the `implementation` set, so it's `undefined`.

Reply:
We've decided to apply the previous assertion, where `undefined` value for `implementation` means the config API is supported.